### PR TITLE
Add support for App Distribution in-app Feedback

### PR DESCRIPTION
**Pull request by kaibolay**
Adding `InAppFeedbackPayload` and `onNewInAppFeedbackPublished()`.

### CODE REVIEW

File: src/v2/providers/alerts/appDistribution.ts [hunk 1]
```diff
@@ -234,6 +265,79 @@ export function onNewTesterIosDevicePublished(
...
+): CloudFunction<AppDistributionEvent<InAppFeedbackPayload>>;
+
+/**
+ * Declares a function that can handle receiving new in-app feedback from a tester.
+ * @param appId - A specific application the handler will trigger on.
+ * @param handler - Event handler which is run every time new feedback is received.
+ * @returns A function that you can export and deploy.
+ */
+export function onInAppFeedbackPublished(
+  appId: string,
```
Comment (colerogers) [inline src/v2/providers/alerts/appDistribution.ts:286]: Just to confirm, `appId` is optional when registering on Eventarc, right?

**Review by colerogers — COMMENTED** [review]
Is this live on the backend? If not, do you know when I can try this branch out?

File: src/v2/providers/alerts/appDistribution.ts [hunk 2]
```diff
@@ -59,6 +89,7 @@ export interface AppDistributionEvent<T>
 
 /** @internal */
 export const newTesterIosDeviceAlert = 'appDistribution.newTesterIosDevice';
+export const inAppFeedbackAlert = 'appDistribution.inAppFeedback';
```
[the remark this answers is not in the document]
Comment (kaibolay) [reply src/v2/providers/alerts/appDistribution.ts:93]: Done.

File: src/v2/providers/alerts/appDistribution.ts (hunk 1 above)
Comment (kaibolay) [reply src/v2/providers/alerts/appDistribution.ts:287]: Yes. It's the same as for `newTesterIosDevice`. Though I have to admit I didn't test this for either yet.

I guess we can test this as soon as http://cl/459802137 is rolled out to production. Do you have a good guesstimate when that might happen?

**Comment by kaibolay** [comment]
> Is this live on the backend? 

I don't think so. I think it depends on http://cl/459802137 making it to production.

 > If not, do you know when I can try this branch out?

I had hoped you could tell me ;-)  I'm not aware of any other missing pieces - please let me know if you can think of any.

File: src/v2/providers/alerts/appDistribution.ts (hunk 1 above)
Comment (colerogers) [reply src/v2/providers/alerts/appDistribution.ts:287]: Usually it takes about a week or so for a full rollout

Comment (kaibolay) [reply src/v2/providers/alerts/appDistribution.ts:287]: I was able to deploy successfully.

**Review by colerogers — APPROVED** [review]
Just ran a successful deploy, LGTM

File: src/v2/providers/alerts/appDistribution.ts [hunk 3]
```diff
@@ -45,6 +45,36 @@ export interface NewTesterDevicePayload {
...
+  /**
+   * Build version of the release. For an Android release, the build version
+   * is the `versionCode`. For an iOS release, the build version is the
+   * `CFBundleVersion`.
+   */
+  buildVersion: string;
+  /** Text entered by the tester */
+  text: string;
+  /** URIs to download screenshot(s) */
+  screenshotUris?: string[];
```
Comment (inlined) [inline src/v2/providers/alerts/appDistribution.ts:75]: Can we make this non-optional and always provide the empty array if necessary?

Comment (colerogers) [reply src/v2/providers/alerts/appDistribution.ts:75]: In the SDK, correct?

Comment (jladieu) [reply src/v2/providers/alerts/appDistribution.ts:75]: Hi there, I'm on Kai's team and working on this while he's on vacation.  

https://github.com/firebase/firebase-functions/pull/1187 includes a fix for this feedback.

Comment (inlined) [reply src/v2/providers/alerts/appDistribution.ts:75]: No worries. I helped out in that PR.

**Pull request merged**